src/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointListener.cs

```diff
@@ -56,7 +56,11 @@ public HttpEndPointListener(HttpListener listener, IPAddress addr, int port, boo
             {
                 _secure = secure;
                 // TODO #14691: Implement functionality to read SSL certificate.
+#if MONO
+                _cert = _listener.LoadCertificateAndKey (addr, port);
```

Comment:
Sure, I can add a new partial class.  This will also makes it easier to provide a new, general-purpose implementation for CoreFx.